journal.py

```diff
@@ -51,36 +67,83 @@ def all(cls, session=None):
             session = DBSession
         return session.query(cls).order_by(cls.date.desc()).all()
 
+    @classmethod
+    def search(cls, id, session=None):
+        if session is None:
+            session = DBSession
+        return session.query(cls).filter_by(id=id).one()
+
+    @classmethod
+    def update(cls, id, title=None, text=None, session=None):
+        if session is None:
+            session = DBSession
+        entry = session.query(cls).filter_by(id=id).one()
+        entry.title = title
+        entry.text = text
+        return entry
+
 
 @view_config(route_name='home', renderer='templates/index.jinja2')
-def _view(request):
+def index_view(request):
     entries = Entry.all()
-    return {'entries':entries}
+    return {'entries': entries, 'add_markdown': add_markdown}
 
 
 @view_config(route_name='detail', renderer='templates/detail.jinja2')
 def detail_view(request):
-    entries = Entry.all()
-    return {'entries':entries}
+    post_id = request.matchdict.get('id', None)
+    try:
+            entry = Entry.search(post_id)
+    except NoResultFound:
+        return HTTPNotFound('No post found.')
+    return {'entry': entry, 'add_markdown': add_markdown}
 
 
 @view_config(route_name='edit', renderer='templates/edit.jinja2')
-def edit_view(request):
-    entries = Entry.all()
-    return {'entries':entries}
+def edit_entry(request):
+    if request.authenticated_userid:
+        post_id = request.matchdict.get('id', None)
+        try:
+            entry = Entry.search(post_id)
+        except NoResultFound:
+            return HTTPNotFound('No post found.')
+        if request.method == 'POST':
+            id = request.matchdict['id']
+            title = request.params.get('title')
+            text = request.params.get('text')
+            Entry.update(id=id, title=title, text=text)
```

Comment:
given that you already have the specific entry you are editing bound to the symbol `entry` (see line 107 above), you could just update the title and text attributes here, rather than calling the 'update' method.  

Alternatively, you could implement `update` as an *instance method* instead of as a classmethod.  That would allow you to encapsulate the update functionality, but use the `entry` you've got instead of looking it up in the database a second time.